packages/node-core/src/utils/autoQueue.ts

```diff
@@ -169,6 +168,9 @@ export class AutoQueue<T> implements IQueue {
     // Empty the queue
     // TODO do we need to reject all promises?
     this.queue.takeAll();
+    this.pendingPromise = false;
+    this._abort = false;
+    this.processingTasks = 0;
```

Comment:
Im not too sure how this fixes the issue. If we empty the queue, then wont this happen https://github.com/subquery/subql/blob/40edb57698f35f42ae567edc087954576d4a0a4e/packages/node-core/src/utils/autoQueue.ts#L145 and these values get set anyway?

I feel like the test would still pass without these changes.